pkg/river/internal/value/decode.go

```diff
@@ -64,15 +64,12 @@ func decode(val Value, into reflect.Value) error {
 		into = into.Elem()
 	}
 
-	// Fastest cases: we can directly assign values.
+	// Fastest cases: we can directly assign values without converting.
 	switch {
 	case val.Type() == TypeNull:
 		// TODO(rfratto): Does it make sense for a null to always decode into the
 		// zero value? Maybe only objects and arrays should support null?
 		into.Set(reflect.Zero(into.Type()))
-	case val.rv.Type() == into.Type():
-		into.Set(cloneGoValue(val.rv))
-		return nil
```

Comment:
I had to remove this because it led to too much unpredictable decoding, and was causing tests for `concat` function to fail, which were expecting `int`s from decodeAny, but that path wasn't getting hit because of this check. 

We might be able to add it back in the future, but removing it is easier for now and gives me more confidence in knowing what the result of decoding is.